docs/ref/middleware.txt

```diff
@@ -495,6 +495,19 @@ Adds the ``user`` attribute, representing the currently-logged-in user, to
 every incoming ``HttpRequest`` object. See :ref:`Authentication in web requests
 <auth-web-requests>`.
 
+.. class:: LoginRequiredMiddleware
+
+.. versionadded:: 5.1
+
+By default, this middleware mandates login for all views. However, views
+employing the ``login_not_required`` decorator or the
+``LoginNotRequiredMixin`` mixin are exempt from this requirement. In the
+absence of these, the middleware will guide the user to the login page or to
+a custom login page if the view has defined ``login_url`` or
+``redirect_field_name`` attributes. This is typically seen when using the
+``login_required`` decorator or the ``LoginRequiredMixin`` mixin and
+modifying the default behavior.
```

Comment:
Maybe something like:
```
Redirects all requests by non-authenticated users to a login page. For admin 
views, this will redirect to the admin login. For all other views, this will 
redirect to :setting:`settings.LOGIN_URL <LOGIN_URL>`. This can be customized 
by setting ``login_url`` or ``redirect_field_name`` attributes on your view.
```
And then we can insert an example of doing this for a class based and function based view. Something like

```python
class MyView(View):
     login_url = "/login/"
     redirect_field_name = "redirect_to"
```
and....? Not sure if this is right but
```python
@login_required(login_url="/accounts/login/")
def my_view(request): ...
```
and then finish with
```
Views using the :func:`~django.contrib.auth.decorators.login_not_required` 
decorator or the :class:`~django.contrib.auth.mixins.LoginNotRequiredMixin` 
mixin are exempt from this requirement.
```